src/framework/vst/internal/vstconfiguration.cpp

```diff
@@ -1,43 +1,44 @@
-//  it under the terms of the GNU General Public License version 2.
-//
-//  This program is distributed in the hope that it will be useful,
-//  but WITHOUT ANY WARRANTY; without even the implied warranty of
-//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-//  GNU General Public License for more details.
-//
-//  You should have received a copy of the GNU General Public License
-//  along with this program; if not, write to the Free Software
-//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
-//=============================================================================
-#include "vstconfiguration.h"
-#include "settings.h"
-
-using namespace mu::vst;
-using namespace mu::framework;
-
-const Settings::Key VSTConfiguration::SEARCH_PATHS = Settings::Key("vst", "search_path");
-
-#ifdef Q_OS_MAC
-const std::string VSTConfiguration::DEFAULT_PATHS = "/Library/Audio/Plug-Ins/VST3";
-#elif defined(Q_OS_WIN)
-const std::string VSTConfiguration::DEFAULT_PATHS = "C:\Program Files (x86)\Common Files\VST3";
-#else
-const std::string VSTConfiguration::DEFAULT_PATHS = "";
-#endif
-
-void VSTConfiguration::init()
-{
-    settings()->setDefaultValue(SEARCH_PATHS, Val(VSTConfiguration::DEFAULT_PATHS));
-}
-
-std::string VSTConfiguration::searchPaths() const
-{
-    return settings()->value(SEARCH_PATHS).toString();
-}
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2021 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "vstconfiguration.h"
+#include "settings.h"
+
+using namespace mu::vst;
+using namespace mu::framework;
+
+static const Settings::Key CUSTOM_SEARCH_PATH_KEY = Settings::Key("vst", "custom_search_path");
+
+mu::io::path VstConfiguration::customSearchPath() const
+{
+    return mu::io::path(settings()->value(CUSTOM_SEARCH_PATH_KEY).toString());
+}
+
+Steinberg::FIDString VstConfiguration::currentPlatformType() const
+{
+#ifdef Q_OS_MAC
+    return Steinberg::kPlatformTypeNSView;
+#elif defined(Q_OS_IOS)
+    return Steinberg::kPlatformTypeUIView;
+#elif defined(Q_OS_WIN)
+    return Steinberg::kPlatformTypeHWND;
+#elif defined(Q_OS_LINUX)
+    return Steinberg::kPlatformTypeX11EmbedWindowID;
+#endif

```

Comment:
It would be great to add some default return here

Reply:
I think it was a bad idea to put this method here, anyway it worth to add default value. Thx